### PR TITLE
chore(actor) Backfill rule.owner fields

### PR DESCRIPTION
**Pull request by markstory**
As part of the work to remove `Actor` we need to populate `owner_user_id` and `owner_team_id` on `Rule`.

New writes were addressed with #69427

### CODE REVIEW

File: src/sentry/migrations/0704_backfill_rule_user_team.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        actor = Actor.objects.get(id=rule.owner_id)
+        changed = False
+        if actor.user_id:
+            rule.owner_user_id = actor.user_id
+            changed = True
+        elif actor.team_id:
+            rule.owner_team_id = actor.team_id
+            changed = True
+        else:
+            click.echo(f"Actor {actor.id} is neither a user or team")
```
Comment (wedamija) [inline src/sentry/migrations/0704_backfill_rule_user_team.py:30]: Is using click ok in a migration?

**Review by GabeVillalobos — APPROVED** [review]
🎉 Nice work!

**Review by wedamija — APPROVED** [review]

File: src/sentry/migrations/0704_backfill_rule_user_team.py (hunk 1 above)
Comment (markstory) [reply src/sentry/migrations/0704_backfill_rule_user_team.py:30]: I hope so, because print() is banned by our flake8 rules.

Comment (wedamija) [reply src/sentry/migrations/0704_backfill_rule_user_team.py:30]: Should we use logging instead?

**Comment by wedamija** [comment]
> ## Suspect Issues
> This pull request was deployed and Sentry observed the following issues:
> 
> * ‼️ **OperationalError: QueryCanceled('canceling statement due to user request\n')** `sentry.db.postgres.base in execute` [View Issue](https://sentry.sentry.io/issues/5249778451/?referrer=github-pr-bot)
> 
> Did you find this useful? React with a 👍 or 👎

@markstory I guess this is timing out when running due to the filters when using the `RangeQuerySetWrapper`. Might need to use `.all()` there and filter in memory

**Comment by markstory** [comment]
> I guess this is timing out when running due to the filters when using the RangeQuerySetWrapper. Might need to use .all() there and filter in memory

I re-ran the migration a [second time](https://deploy.getsentry.net/go/pipelines/post-deploy-migrations/79/run-migrations/2/jobs) and didn't hit the same timeout condition. If it had failed a second time, I agree that using .all() and doing filtering in memory would have been the best path forward.

**Pull request merged**